client/src/components/App.js

```diff
@@ -7,14 +7,45 @@ import Header from './Header';
 import MapContainer from './MapContainer';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedOption: '',
+      defaultCenter: { lat: 31.771959, lng: 35.217018 },
+      defaultZoom: 8,
+    };
+    this.handleSearch = this.handleSearch.bind(this);
+  }
+
+  handleSearch(coord, callback) {
+    this.setState({ defaultCenter: coord, defaultZoom: 12 }, () => {
+      callback();
```

Comment:
i dont understand this callback do you need it here, can you try this code without having the callback?

Reply:
correct me if im wrong but after the state changes , you want to change to a different page ?

Reply:
The callback allows me to redirect the page only after the state has changed. I think it's better that way.